client/src/main/java/io/pravega/client/tables/TableTransaction.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import com.google.common.annotations.Beta;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines all operations that can be made on a Table Transaction.
+ *
+ * Notes about Updates:
+ * * Updates (Conditional or not) in this TableTransaction will not be reflected in the main Table until {@link #commit()}
+ * is invoked (and completes successfully).
+ * * Conditional Updates in this TableTransaction can only be done by comparing with Keys in this Transaction. Comparisons
+ * cannot be made against Keys from the main Table.
+ * * {@link #commit()} cannot be made conditional against the state of the main Table.
+ *
+ * @param <KeyT>   Table Key Type.
+ * @param <ValueT> Table Value Type.
+ */
+@Beta
+public interface TableTransaction<KeyT, ValueT> extends AutoCloseable {
+    /**
+     * Performs an Unconditional Update by inserting or updating the Value for a Key in this Table, regardless of whether
+     * the Key exists or not or what its Version is. See {@link TableWriter#put(KeyT, ValueT)}.
+     *
+     * @param key   The Key to insert or update.
+     * @param value The Value to set for the Key.
+     * @return A CompletableFuture that, when completed, will contain the {@link KeyVersion} associated with the newly
+     * inserted or updated entry.
+     */
+    CompletableFuture<KeyVersion> put(KeyT key, ValueT value);
```

Comment:
This conditional update is within the scope of the Table Transaction. Perhaps someone wants to condition an update based on the keys already present in that transaction? 

You are correct in that this cannot be based off the main table, and it can only check against the table transaction itself.